imkar/scattering/scattering.py

```diff
@@ -0,0 +1,149 @@
+import numpy as np
+import pyfar as pf
+from imkar import utils
+
+
+def freefield(sample_pressure, reference_pressure, microphone_weights):
+    r"""
+    Calculate the direction dependent free-field scattering coefficient.
+
+    Uses the Mommertz correlation method [1]_ to calculate the scattering
+    coefficient of the input data:
+
+    .. math::
+        s = 1 -
+            \frac{|\sum_w \underline{p}_{\text{sample}}(\vartheta,\varphi)
+            \cdot \underline{p}_{\text{reference}}^*(\vartheta,\varphi)
+            \cdot w(\vartheta,\varphi)|^2}
+            {\sum_w |\underline{p}_{\text{sample}}(\vartheta,\varphi)|^2
+            \cdot w(\vartheta,\varphi) \cdot \sum_w
+            |\underline{p}_{\text{reference}}(\vartheta,\varphi)|^2
+            \cdot w(\vartheta,\varphi) }
+
+    with the reflected sound pressure of the the sample under investigation
+    :math:`\underline{p}_{\text{sample}}`, the reflected sound pressure from
+    the reference sample (same dimension as the sample under investigation,
+    but with flat surface) :math:`\underline{p}_{\text{reference}}`, the
+    area weights of the sampling :math:`w`, and :math:`\vartheta` and
+    :math:`\varphi` are the incidence angle and azimuth angles. See
+    :py:func:`random` to calculate the random incidence
+    scattering coefficient.
+
+    Parameters
+    ----------
+    sample_pressure : :py:class:`~pyfar.classes.audio.FrequencyData`
+        Reflected sound pressure or directivity of the test sample. Its cshape
+        needs to be (..., microphone_weights.csize).
+    reference_pressure : :py:class:`~pyfar.classes.audio.FrequencyData`
+        Reflected sound pressure or directivity of the
+        reference sample. Needs to have the same cshape and frequencies as
```

Comment:
Should we link to pyfar audio concepts when refering to `cshape`? Otherwise people less familiar with pyfar might be confused.